src/main/java/org/ibp/api/java/impl/middleware/program/ProgramServiceImpl.java

```diff
@@ -47,7 +58,28 @@
 	@Autowired
 	private RemoveProgramMembersValidator removeProgramMembersValidator;
 
+	@Autowired
+	private LocationService locationService;
+
+	@Autowired
+	private ProgramFavoriteService programFavoriteService;
+
+	@Autowired
+	private ProgramBasicDetailsDtoValidator programBasicDetailsDtoValidator;
+
+	@Autowired
+	private StudyService studyService;
+
+	@Autowired
+	private MethodService methodService;
+
+	@Autowired
+	private GermplasmListService germplasmListService;
+
+	private final InstallationDirectoryUtil installationDirectoryUtil = new InstallationDirectoryUtil();
+
 	public static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd");
+	public static final String UNSPECIFIED_LOCATION = "Unspecified Location";
```

Comment:
There's another one in org.generationcp.middleware.pojos.Location#UNSPECIFIED_LOCATION, maybe we can use that